cpp/src/fil/infer.cu

```diff
@@ -308,18 +327,18 @@ struct tree_aggregator_t<NITEMS, CATEGORICAL_LEAF> {
   int* votes;
   int num_classes;
 
-  static size_t smem_finalize_footprint(int num_classes) {
-    return sizeof(int) * num_classes * NITEMS;
+  static size_t smem_finalize_footprint(shmem_size_params params) {
+    return smem_accumulate_footprint(params.num_classes) +
+           cols_shmem_size(params);
   }
   static size_t smem_accumulate_footprint(int num_classes) {
-    return smem_finalize_footprint(num_classes);
+    return sizeof(int) * num_classes * NITEMS;
   }
 
-  __device__ __forceinline__ tree_aggregator_t(int num_classes_,
-                                               void* shared_workspace,
-                                               size_t data_row_size)
-    : num_classes(num_classes_),
-      votes((int*)(data_row_size + (char*)shared_workspace)) {
+  __device__ __forceinline__ tree_aggregator_t(predict_params params,
+                                               void* accumulate_workspace,
+                                               void* finalize_workspace)
+    : num_classes(params.ssp.num_classes), votes((int*)accumulate_workspace) {
```

Comment:
e.g. the `votes((int*)accumulate_workspace)` case here is obviously correct, and gives a hint that `votes` are used during accumulation